runtime-light/server/http/init-functions.cpp

```diff
@@ -0,0 +1,188 @@
+    server.set_value(string{ARGV, std::char_traits<char>::length(ARGV)}, invoke_http.uri.opt_query.value_or(string{}));
+  }
+  server.set_value(string{PHP_SELF, std::char_traits<char>::length(PHP_SELF)}, invoke_http.uri.path);
+  server.set_value(string{SCRIPT_NAME, std::char_traits<char>::length(SCRIPT_NAME)}, invoke_http.uri.path);
+  server.set_value(string{SCRIPT_URL, std::char_traits<char>::length(SCRIPT_URL)}, invoke_http.uri.path);
+
+  server.set_value(string{SERVER_ADDR, std::char_traits<char>::length(SERVER_ADDR)}, invoke_http.connection.server_addr);
+  server.set_value(string{SERVER_PORT, std::char_traits<char>::length(SERVER_PORT)}, static_cast<int64_t>(invoke_http.connection.server_port));
+  server.set_value(string{SERVER_PROTOCOL, std::char_traits<char>::length(SERVER_PROTOCOL)},
+                   get_server_protocol(invoke_http.version, invoke_http.uri.opt_scheme));
+  server.set_value(string{REMOTE_ADDR, std::char_traits<char>::length(REMOTE_ADDR)}, invoke_http.connection.remote_addr);
+  server.set_value(string{REMOTE_PORT, std::char_traits<char>::length(REMOTE_PORT)}, static_cast<int64_t>(invoke_http.connection.remote_port));
+
+  server.set_value(string{REQUEST_METHOD, std::char_traits<char>::length(REQUEST_METHOD)}, invoke_http.method);
+  server.set_value(string{GATEWAY_INTERFACE, std::char_traits<char>::length(GATEWAY_INTERFACE)},
+                   string{GATEWAY_INTERFACE_VALUE, std::char_traits<char>::length(GATEWAY_INTERFACE_VALUE)});
+
+  if (invoke_http.uri.opt_scheme.has_value() && std::strcmp((*invoke_http.uri.opt_scheme).c_str(), HTTPS_SCHEME) == 0) {
+    server.set_value(string{HTTPS, std::char_traits<char>::length(HTTPS)}, true);
+  }
+
+  if (invoke_http.uri.opt_query.has_value()) {
+    // TODO v_GET
+    server.set_value(string{QUERY_STRING, std::char_traits<char>::length(QUERY_STRING)}, *invoke_http.uri.opt_query);
+
+    string uri_string{};
+    uri_string.reserve_at_least((invoke_http.uri.path.size()) + (*invoke_http.uri.opt_query).size() + 1); // +1 for '?'
+    uri_string.append(invoke_http.uri.path);
+    uri_string.append(1, '?');
+    uri_string.append(*invoke_http.uri.opt_query);
+    server.set_value(string{REQUEST_URI, std::char_traits<char>::length(REQUEST_URI)}, uri_string);
+  } else {
+    server.set_value(string{REQUEST_URI, std::char_traits<char>::length(REQUEST_URI)}, invoke_http.uri.path);
+  }
+
+  process_headers(std::move(invoke_http.headers), superglobals);
+  // TODO connection_kind, CONTENT_TYPE, is_head_query, _REQUEST, v_POST
+
+  string real_scheme_header{HTTP_X_REAL_SCHEME, std::char_traits<char>::length(HTTP_X_REAL_SCHEME)};
+  string real_host_header{HTTP_X_REAL_HOST, std::char_traits<char>::length(HTTP_X_REAL_HOST)};
+  string real_request_header{HTTP_X_REAL_REQUEST, std::char_traits<char>::length(HTTP_X_REAL_REQUEST)};
+  if (server.isset(real_scheme_header) || server.isset(real_host_header) || server.isset(real_request_header)) {
```

Comment:
In [`runtime/interface.cpp`](https://github.com/VKCOM/kphp/blob/master/runtime/interface.cpp#L1680) condition is different

Reply:
Can you please provide a link or an example?

Reply:
You're right, fixed